packages/collaboration/src/browser/collaboration-frontend-contribution.ts

```diff
@@ -0,0 +1,321 @@
+import { Deferred } from '@theia/core/lib/common/promise-util';
+import { CollaborationWorkspaceService } from './collaboration-workspace-service';
+import { StatusBar, StatusBarAlignment, StatusBarEntry } from '@theia/core/lib/browser/status-bar';
+import { codiconArray } from '@theia/core/lib/browser/widgets/widget';
+import { FrontendApplicationConfigProvider } from '@theia/core/lib/browser/frontend-application-config-provider';
+
+export const COLLABORATION_CATEGORY = 'Collaboration';
+
+export namespace CollaborationCommands {
+    export const CREATE_ROOM: Command = {
+        id: 'collaboration.create-room'
+    };
+    export const JOIN_ROOM: Command = {
+        id: 'collaboration.join-room'
+    };
+}
+
+export const COLLABORATION_STATUS_BAR_ID = 'statusBar.collaboration';
+
+export const COLLABORATION_AUTH_TOKEN = 'THEIA_COLLAB_AUTH_TOKEN';
+export const COLLABORATION_SERVER_URL = 'COLLABORATION_SERVER_URL';
+export const DEFAULT_COLLABORATION_SERVER_URL = 'https://oct-server-staging-ymijt5gjsa-ew.a.run.app/';
+
+@injectable()
+export class CollaborationFrontendContribution implements CommandContribution {
+
+    protected readonly authHandlerDeferred = new Deferred<ConnectionProvider>();
+
+    @inject(WindowService)
+    protected readonly windowService: WindowService;
+
+    @inject(QuickInputService) @optional()
+    protected readonly quickInputService?: QuickInputService;
+
+    @inject(EnvVariablesServer)
+    protected readonly envVariables: EnvVariablesServer;
+
+    @inject(CollaborationWorkspaceService)
+    protected readonly workspaceService: CollaborationWorkspaceService;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    @inject(CommandRegistry)
+    protected readonly commands: CommandRegistry;
+
+    @inject(StatusBar)
+    protected readonly statusBar: StatusBar;
+
+    @inject(CollaborationInstanceFactory)
+    protected readonly collaborationInstanceFactory: CollaborationInstanceFactory;
+
+    protected currentInstance?: CollaborationInstance;
+
+    @postConstruct()
+    protected init(): void {
+        this.setStatusBarEntryDefault();
+        this.getCollaborationServerUrl().then(serverUrl => {
+            const authHandler = new ConnectionProvider({
+                url: serverUrl,
+                client: FrontendApplicationConfigProvider.get().applicationName,
+                fetch: window.fetch.bind(window),
+                opener: url => this.windowService.openNewWindow(url),
+                transports: [SocketIoTransportProvider],
+                userToken: localStorage.getItem(COLLABORATION_AUTH_TOKEN) ?? undefined
+            });
+            this.authHandlerDeferred.resolve(authHandler);
+        }, err => this.authHandlerDeferred.reject(err));
+    }
+
+    protected async onStatusDefaultClick(): Promise<void> {
+        const items: QuickPickItem[] = [];
+        if (this.workspaceService.opened) {
```

Comment:
`opened` in this case doesn't mean that the explorer node is expanded. It just means that there is a workspace available (or not if the value is `false`). Since there is nothing to share if we don't have a workspace open, we don't show the option.